libtransmission/rpc-server.cc

```diff
@@ -587,38 +705,48 @@ static void startServer(void* vserver)
         return;
     }
 
-    struct evhttp* httpd = evhttp_new(server->session->event_base);
+    struct event_base* base = server->session->event_base;
+    struct evhttp* httpd = evhttp_new(base);
+
     evhttp_set_allowed_methods(httpd, EVHTTP_REQ_GET | EVHTTP_REQ_POST | EVHTTP_REQ_OPTIONS);
 
     char const* address = tr_rpcGetBindAddress(server);
 
     tr_port const port = server->port;
 
-    if (evhttp_bind_socket(httpd, address, port) == -1)
+    bool success = false;
+    if (server->bindAddress.type == TR_RPC_AF_UNIX)
+    {
+        success = bindUnixSocket(base, httpd, address, server->rpc_socket_mode);
+    }
+    else
+    {
+        success = evhttp_bind_socket(httpd, address, port) != -1;
+    }
+
+    char addr_port_str[TrUnixAddrStrLen + 7];
```

Comment:
This `TrUnixAddrStrLen + 7` idiom shows up three times and is confusing. Maybe this deserves its own symbolic name to explain what the +7 is for.

Optional, but maybe just remove the `_to_string()` off the end of that function name and have it return a std::string so that the callers don't have to know the maxlen in advance? `tr_rpc_address_with_port_to_string` is only ever going to be called a handful of times in a session so the overhead of a few std::strings is not bad

Reply:
This is a leftover of the previous implementation which I probably should have changed, the 7 is suppoed to be the largest port, 65535, plus `:` plus `\0`. However, that doesn't make sense anymore. Changing this to `std::string` should be possible.